grand_challenge_dicom_de_identifier/deidentifier.py

```diff
@@ -255,20 +265,21 @@ def set_deidentification_method_tag(self, dataset: Dataset) -> None:
         """
         version = self.procedure.get("version", "unknown")
         timestamp = datetime.now().isoformat()
-        method = (
-            f"De-identified by Python DICOM de-identifier using procedure "
-            f"version {version} on {timestamp}."
-        )
 
-        existing_method = dataset.get("DeidentificationMethod", "")
-        if existing_method:
-            new_method = f"{existing_method}; {method}"
+        # Ensure DeidentificationMethodCodeSequence exists and is a Sequence
+        if "DeidentificationMethodCodeSequence" in dataset:
+            method_seq = cast(
+                pydicom.sequence.Sequence,
+                dataset.DeidentificationMethodCodeSequence,
+            )
         else:
-            new_method = method
+            method_seq = pydicom.sequence.Sequence()
+            dataset.DeidentificationMethodCodeSequence = method_seq
 
-        # DICOM tag (0012,0063) - De-identification Method
-        dataset.add_new(
-            tag=pydicom.tag.Tag(0x0012, 0x0063),
-            VR="LO",
-            value=new_method,
+        code = Dataset()
+        code.CodeMeaning = "Description of method"
+        code.LongCodeValue = (
+            f"De-identified by Python DICOM de-identifier using procedure "
+            f"version {version} on {timestamp}."
         )
+        method_seq.append(code)
```

Comment:
Why would you use this sequence tag instead of 0012,0063? I see how you can make a case that this can be expanded upon more easily when multiple deidentification methods have been run, but I'm not sure if it makes sense to fully remove setting of 0012,0063?